### PR TITLE
Boolean aggs

### PR DESCRIPTION
**Pull request by hlminh2000**
screen shot 2018-03-20 at 2 10 07 pm

### CODE REVIEW

File: modules/components/src/Aggs/BooleanAgg.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    evaluateInSqon: ({ sqon }) =>
+      inCurrentSQON({
+        dotField,
+        value: booleanValues.false,
+        currentSQON: sqon,
+      }),
+  });
+  const isNeitherActive = !isTrueActive && !isFalseActive;
+
+  const handleTrueClick = () =>
```
Comment (alex-wilmer) [inline modules/components/src/Aggs/BooleanAgg.js]: do we really need both of these functions, and not just one function that takes a boolean argument?

**Comment by alex-wilmer** [comment]
Could you include a screenshot of this agg in this pull request please?

File: modules/components/src/Aggs/BooleanAgg.js (hunk 1 above)
Comment (hlminh2000) [reply modules/components/src/Aggs/BooleanAgg.js]: not sure why I kept two... yes, refactored!

File: modules/components/src/Aggs/AggsWrapper.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import Component from 'react-component-component';
+import './AggregationCard.css';
+
+export default ({ children, collapsible = true, displayName }) => (
+  <Component initialState={{ isCollapsed: false, showingMore: false }}>
+    {({ setState, state: { isCollapsed, showingMore } }) => (
+      <div className="test-term-aggregation aggregation-card">
+        <div
+          className={`title-wrapper ${isCollapsed && 'collapsed'}`}
```
Comment (Jephuff) [inline modules/components/src/Aggs/AggsWrapper.js]: make this `className={`title-wrapper ${isCollapsed ? 'collapsed' : ''}`}` or you will end up with a `false` class :)

File: modules/components/src/Aggs/AggsWrapper.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import Component from 'react-component-component';
+import './AggregationCard.css';
+
+export default ({ children, collapsible = true, displayName }) => (
+  <Component initialState={{ isCollapsed: false, showingMore: false }}>
+    {({ setState, state: { isCollapsed, showingMore } }) => (
+      <div className="test-term-aggregation aggregation-card">
```
Comment (Jephuff) [inline modules/components/src/Aggs/AggsWrapper.js]: not just for term aggregations?

File: modules/components/src/Aggs/AggsWrapper.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+              : () => {}
+          }
+        >
+          <span className="title">{displayName}</span>
+          {collapsible && (
+            <span className={`arrow ${isCollapsed && 'collapsed'}`} />
+          )}
+        </div>
+        {!isCollapsed && (
+          <div className={`bucket ${isCollapsed && 'collapsed'}`}>
```
Comment (Jephuff) [inline modules/components/src/Aggs/AggsWrapper.js]: false class here too

File: modules/components/src/Aggs/BooleanAgg.css [hunk 5]
```diff
@@ -0,0 +1,38 @@
+.booleanFacetWrapper {
+  display: flex;
+  flex-direction: row;
+  height: 30px;
+  font-size: 0.9em;
+}
+.booleanFacetAny {
```
Comment (Jephuff) [inline modules/components/src/Aggs/BooleanAgg.css]: a lot of shared styles between these. Could be good to make one selector for the general stuff

File: modules/components/src/Aggs/BooleanAgg.js [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+const booleanValues = {
+  true: 'true',
+  false: 'false',
+};
+
+export default ({
+  field,
+  buckets,
+  handleValueClick = () => {},
+  isTrueSelected = () => false,
```
Comment (Jephuff) [inline modules/components/src/Aggs/BooleanAgg.js]: can this match the `isActive` api in the term facet? seems like it's doing the same thing

File: modules/components/src/Aggs/BooleanAgg.js [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+      generateNextSQON: sqon =>
+        replaceSQON(
+          {
+            op: 'and',
+            content: [
+              {
+                op: 'in',
+                content: {
+                  field: dotField,
+                  value: [isTrue ? booleanValues.true : booleanValues.false],
```
Comment (Jephuff) [inline modules/components/src/Aggs/BooleanAgg.js]: what happens if you don't convert this to a string?

File: modules/components/src/Aggs/AggsWrapper.js [hunk 8]
```diff
@@ -0,0 +1,30 @@
+import React from 'react';
```
Comment (Jephuff) [inline modules/components/src/Aggs/AggsWrapper.js]: Could you use this in the range and term agg? If it doesn't happen now it will never happen :D

Comment (hlminh2000) [reply modules/components/src/Aggs/AggsWrapper.js]: done! :D had that as a todo for the AFV rewrite actually, but done anyway :)

Comment (Jephuff) [reply modules/components/src/Aggs/AggsWrapper.js]: ahh fair :)

File: modules/components/src/Aggs/BooleanAgg.js [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+
+  const createSqonValidator = isTrue => ({ sqon }) =>
+    inCurrentSQON({
+      dotField,
+      value: isTrue,
+      currentSQON: sqon,
+    });
+
+  const isTrueActive = isActive({
+    defaultEvaluator: createSqonValidator(true),
```
Comment (Jephuff) [inline modules/components/src/Aggs/BooleanAgg.js]: I don't think you need `defualtEvaluator` or `createSqonValidator`. the term and boolean agg do the same thing for `isActive` so just let the parent handle this.

**Pull request merged**